src/recipes/tooling-custom-resources.md

```diff
@@ -0,0 +1,40 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Custom Resources
+
+Custom `Resource`s are exposed for the end user to use within their development, but cannot provide any functionality automatically.
+For example, you can create a custom `AudioStream` type that handles a new and interesting audio file type.
+
+
+## Registering a Resource
+
+This workflow is incredibly similar to the [Hello World Example](/intro/hello-world.md)
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, init, base=Resource)] // use tool!
+struct ResourceType {
+    #[base]
+    base: Base<Resource>,
+}
+
+#[godot_api]
+impl IResource for ResourceType {}
```

Comment:
Empty `impl` blocks are not needed. Mentioning them in code samples might give the impression that they are, as it was the case before I implemented https://github.com/godot-rust/gdext/pull/500.